shell/platform/android/io/flutter/plugin/platform/PlatformView.java

```diff
@@ -24,4 +25,26 @@
      * after this method is called. Failing to do so will result in a memory leak.
      */
     void dispose();
+
+    /**
+    * Callback fired when the platform's input connection is locked, or should be used. See also
+    * {@link TextInputPlugin#lockPlatformViewInputConnection}.
+    *
+    * <p>This hook only exists for rare cases where the plugin relies on the state of the input
+    * connection. This probably doesn't need to be implemented.
+    */
+    // Default interface methods are supported on all min SDK versions of Android.
+    @SuppressLint("NewApi")
+    default void onInputConnectionLocked() {};
```

Comment:
[not sure about this, what do you think?]

Alternatively we could call these lockInputConnection/unlockInputConnection, these hooks allows us to move the locking logic out of TextInputPlugin and into the keyboard using platform view, in which case I think the alternative names are better. OTOH I'm not sure if we don't want to provide the locking behavior by default as this should work for any platform view that doesn't do the crazy threading tricks webview is doing.

Reply:
I don't think I understand what you're suggesting, can you go into more detail? Are you saying we should implement logic here by default or is this comment talking about moving this call into `PlatformViewsController#resize`?

Assuming you're talking about shifting logic here, into the `default` methods, I'd rather avoid that if possible. I think doing that would have similar readability issues with base classes, where there's secret behavior up the hierarchy chain that's kind of hard to understand and easily broken accidentally.

Reply:
I'm talking about not doing any locking in the engine code, but leave it up to the plugin.